src/main/java/com/mathworks/ci/MatlabRunTestsStepExecution.java

```diff
@@ -46,10 +46,8 @@ public boolean start() throws Exception {
 
         getContext().setResult((res == 0) ? Result.SUCCESS : Result.FAILURE);
         
-        getContext().onSuccess(true);
-        
         //return false represents the asynchronous run. 
-        return false;
+        return null;
```

Comment:
My Java-Fu is quite old, but why is the `Void` have a capital V here? And also why `return null` over just `return`?

Reply:
Its just alias for actual void. SynchronousNonBlockingStepExecution uses parameters for return types i choose not to return anything here hence its [Void](https://docs.oracle.com/javase/7/docs/api/java/lang/Void.html#:~:text=Class%20Void&text=The%20Void%20class%20is%20an,representing%20the%20Java%20keyword%20void.) since Void is uninstantiable object i just return null.

Reply:
Sounds good. Now I see the generic parameter in the extends line. Tests? What failing test would have alerted us to this?

Reply:
We will have to add integration-test for this which would run against actual MATLAB instance. Unit tests may be a challenge though as nothing on UI or log level that could be verified. I can rather keep a test along side unit tests and make it disabled for unit run.

Reply:
@SangaviJayagopi Lets include this as a requirement for MATLAB in the loop integration testing